mailet/standard/src/main/java/org/apache/james/transport/mailets/RemoveMimeHeaderByPrefix.java

```diff
@@ -70,21 +72,25 @@ public void service(Mail mail) throws MessagingException {
         for (String headerName: headerNamesToRemove) {
             mail.getMessage().removeHeader(headerName);
         }
-        if (!headerNamesToRemove.isEmpty()) {
+
+        boolean specificHeadersRemoved = removeSpecific(mail);
+
+        if (!headerNamesToRemove.isEmpty() || specificHeadersRemoved) {
             mail.getMessage().saveChanges();
         }
-
-        removeSpecific(mail);
```

Comment:
not test for this bug fix?

Reply:
Thank to your explanation.
It was not a bug. the saveChanges method update only the top level headers.